api/system/services/System.js

```diff
@@ -1,8 +1,106 @@
 'use strict';
+const fetch = require('node-fetch');
+
 
 /**
  * Read the documentation (https://strapi.io/documentation/3.0.0-beta.x/guides/services.html#core-services)
  * to customize this service
  */
 
-module.exports = {};
+module.exports = {
+  /**
+   * Promise to add record
+   *
+   * @return {Promise}
+   */
+
+  create: async (values) => {
+    let edsmData = await strapi.services.system.edsmUpdate(values);
+
+    return strapi.query('System').create(edsmData);
+  },
+
+  /**
+   * Promise to edit record
+   *
+   * @return {Promise}
+   */
+
+  update: async (params, values) => {
+
+    if (!values.edsmCoordLocked){
+      let oldData = await strapi.services.system.findOne({id: params.id});
+      oldData = oldData.toJSON();
+
+      if (oldData.edsmCoordLocked == false || oldData.edsmCoordLocked == undefined) {
+        if (values.systemName == undefined) {
+          values.systemName = oldData.systemName;
+        }
+        values.missingSkipCount = oldData.missingSkipCount;
+
+        let edsmData = await strapi.services.system.edsmUpdate(values);
+
+        return strapi.query('System').update(params, edsmData);
+      }
+    }
+
+    return strapi.query('System').update(params, values);
+  },
+
+  /**
+   * Promise to fetch a system from EDSM to update a local record
+   *
+   * @return {Promise}
+   */
+  edsmUpdate: async (values) => {
+    const edsmAPISystem = 'https://www.edsm.net/api-v1/system?showId=1&showCoordinates=1&showPrimaryStar=1';
+
+    const getData = async system => {
+      try {
+        const response = await fetch(edsmAPISystem + '&systemName=' + encodeURIComponent(system));
+        const json = await response.json();
+        return json;
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    let newValues = values;
```

Comment:
This is a reference to `values`, not a new object, so you're basically modyfying `values`.

You can "clone" an object by doing something like `Object.assign()` or shorter ` let newValues = { ...values };`. Keep in mind that only primitives are assigned as new values, so for example if you have an object:

```
values = {
    som: false,
    ting: 'foo bar is cool',
    funny: {
        is: 0.75,
        cool: 'string'
    }
}
```

And you do ` let newObj = { ...values };` then your `newObj` will look like that:

```
newObj: {
    som: false,
    ting: 'foo bar is cool',
    funny: // still a reference to values.funny
}
```

Deep cloning objects is a story of its own, but always keep in mind that if you're using a reference then you're mutating the original object.